go/libraries/doltcore/sqle/dfunctions/dolt_merge.go

```diff
@@ -213,6 +232,13 @@ func mergeIntoWorkingSet(ctx *sql.Context, sess *dsess.DoltSession, roots doltdb
 		return ws, noConflictsOrViolations, threeWayMerge, err
 	}
 
+	if !noCommit && ac {
```

Comment:
I'm not sure if we need to switch on the autocommit setting... it's a good question though... 

Looks like you were thinking if autocommit is disabled, we should skip the merge commit? (since dolt_commit creates a db/tx commit implicitly) 

My gut reaction is that we should not have autocommit come into play. My thinking is that it's one less dimension for customers to have to reason about, it's more consistent with the command line, and it seems like customers should expect dolt_merge to work like git_merge and create that merge commit by default. Plus, there is an available workaround where customers could specify `--no-commit` to still get that behavior. 

If you see another perspective, I'm always happy to reconsider and keep discussing if you feel like there's something important here.